junit5/core/src/main/java/org/jboss/arquillian/junit5/ArquillianExtension.java

```diff
@@ -364,4 +354,32 @@ private static ParameterResolutionException createParameterResolutionException(f
         }
         return new ParameterResolutionException(msg, cause);
     }
+
+    private static void throwError(TestResult result) throws Throwable {
+        if (result != null && result.getStatus() != TestResult.Status.PASSED) {
+            if (result.getThrowable() == null) {
+                throw new TestAbortedException(result.getDescription());
+            } else {
+                AssertionFailedError failedAssertion = failedAssertion(result.getThrowable());
+                if (failedAssertion != null) {
+                    throw failedAssertion;
+                } else {
+                    throw result.getThrowable();
+                }
+            }
+        }
+    }
+
+    private static AssertionFailedError failedAssertion(Throwable exception) {
+        if (exception instanceof IdentifiedTestException) {
+            IdentifiedTestException identified = (IdentifiedTestException) exception;
+            Collection<Throwable> exceptions = identified.getCollectedExceptions().values();
+            if (exceptions.stream().allMatch(AssertionFailedError.class::isInstance)) {
+                return exceptions.stream().filter(AssertionFailedError.class::isInstance)
```

Comment:
Any reason why in these checks we shouldn't just be checking for a `java.lang.AssertionError`?